robot/impl/local_robot.go

```diff
@@ -993,6 +981,16 @@ func dialRobotClient(
 func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config) {
 	var allErrs error
 
+	// Sync Packages before reconfiguring rest of robot and resolving references to any packages
+	// in the config.
+	// TODO(RSDK-1849): Make this non-blocking so other resources that do not require packages can run before package sync finishes.
+	// TODO(RSDK-2710) this should really use Reconfigure for the package and should allow itself to check
+	// if anything has changed.
+	err := r.packageManager.Sync(ctx, newConfig.Packages)
+	if err != nil {
+		allErrs = multierr.Combine(allErrs, err)
+	}
+
```

Comment:
moved to be before services validation so that if any of them are expecting packages to exist, they won't error.  I would appreciate if a second person could double check that this should not have any adverse consequences down the road.

Reply:
Makes sense to me and i can't think of any downsides. packages is the only place where the config is changing to affect other services and components. If we add new functionality in the future that conflicts with it, we'll address it then.